File: Sources/AndroidBackend/Kotlin/CustomOnCheckedChangeListener.kt
package dev.swiftcrossui.androidbackend

import android.widget.CompoundButton

class CustomOnCheckedChangeListener(
    private val action: SwiftAction
): CompoundButton.OnCheckedChangeListener {
    override fun onCheckedChanged(
        buttonView: CompoundButton,
        isChecked: Boolean
    ) {
        action.call()
    }
}

File: Sources/AndroidBackend/CustomOnCheckedChangeListener.swift
import SwiftJava
import AndroidKit

@JavaClass(
    "dev.swiftcrossui.androidbackend.CustomOnCheckedChangeListener",
    implements: AndroidKit.CompoundButton.OnCheckedChangeListener.self
)
class CustomOnCheckedChangeListener: JavaObject {
    @JavaMethod
    @_nonoverride convenience init(
        _ action: SwiftAction?,
        environment: JNIEnvironment? = nil
    )
}
